eth-contracts/contracts/service/ServiceProviderFactory.sol

```diff
@@ -325,13 +338,14 @@ contract ServiceProviderFactory is RegistryContract {
         return newStakeAmount;
     }
 
-    function decreaseStake(
-        uint256 _decreaseStakeAmount
-    ) external returns (uint newTotalStake)
+    function requestDecreaseStake(uint _decreaseStakeAmount)
+    external returns (uint newStakeAmount)
     {
-        _requireIsInitialized();
-
-        // Confirm owner has an endpoint
+        // Confirm no request is in flight
+        require(
+            !decreaseRequestPending(msg.sender), 
+            "Request already pending"
+        );
         require(
             spDetails[msg.sender].numberOfEndpoints > 0,
             "Registered endpoint required to decrease stake"
```

Comment:
i believe call outcome was to simply validateAccountStakeBalance here

Reply:
done with the new function validateBalanceInternal, just accepts an input parameter for balance to validate against. this is necessary because we are validating a new target value during decreaseStake